scripts/prepare-release.sh

```diff
@@ -2,30 +2,75 @@
 set -e
 set -o pipefail
 
-PREVIOUS_VERSION=$(grep integreatly-operator deploy/olm-catalog/integreatly-operator/integreatly-operator.package.yaml | awk -F v '{print $2}')
+if [[ -z "$OLM_TYPE" ]]; then
+  OLM_TYPE="integreatly-operator"
+fi
+
+case $OLM_TYPE in
+  "integreatly-operator")
+    PREVIOUS_VERSION=$(grep $OLM_TYPE deploy/olm-catalog/$OLM_TYPE/$OLM_TYPE.package.yaml | awk -F v '{print $2}') || echo "No previous version"
+    ;;
+  "managed-api-service")
+    PREVIOUS_VERSION=$(grep $OLM_TYPE deploy/olm-catalog/$OLM_TYPE/$OLM_TYPE.package.yaml | awk -F v '{print $3}') || echo "No previous version"
+    ;;
+  *)
+    echo "Invalid OLM_TYPE set"
+    echo "Use \"integreatly-operator\" or \"managed-api-service\""
+    exit 1
+    ;;
+esac
+
+if [[ -z "$ORG" ]]; then
+  ORG="integreatly"
+else
+  ORG="$ORG"
+fi
 
 create_new_csv() {
-  operator-sdk generate csv --csv-version "$VERSION" --default-channel --operator-name integreatly-operator --csv-channel=rhmi --update-crds --from-version "$PREVIOUS_VERSION" --make-manifests=false
+
+  if [[ -z "$PREVIOUS_VERSION" ]]
+    then
+      operator-sdk generate csv --csv-version "$VERSION" --default-channel --operator-name "$OLM_TYPE" --csv-channel=rhmi --update-crds --make-manifests=false
+    else
+      operator-sdk generate csv --csv-version "$VERSION" --default-channel --operator-name "$OLM_TYPE" --csv-channel=rhmi --update-crds --from-version "$PREVIOUS_VERSION" --make-manifests=false
+  fi
 }
 
 update_csv() {
-  operator-sdk generate csv --csv-version "$VERSION" --default-channel --operator-name integreatly-operator --csv-channel=rhmi --update-crds --make-manifests=false
+  operator-sdk generate csv --csv-version "$VERSION" --default-channel --operator-name "$OLM_TYPE" --csv-channel=rhmi --update-crds --make-manifests=false
```

Comment:
Do you know if this is run using your local operator-sdk or does it pick up the OPERATOR_SDK in the same way as Makefile?

Reply:
I'm going to say it uses the local operator-sdk , if the script is directly ran. I am not even sure if you were to use the `make release/prepare` command would it pick up the correct version.

Reply:
It picks it up here: https://github.com/integr8ly/integreatly-operator/blob/master/Makefile#L35-L39

Reply:
I think we can mark this as resolved. Call the script through the make target will use the correct operator-sdk while calling the script directly will used the default system version. From what I have seen this is the same for other scripts in the that scripts folder. The issue I can see is. Delorean calls that script directly, would delorean use it's version of the operator-sdk or in the pipelines does the correct version of the operator-sdk get installed?